include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+       *
+       * @param construction_data The data needed for this process.
+       */
+      void
+      reinit(const ConstructionData<dim, spacedim> &construction_data);
+
+      /**
+       * @note This function is not implemented for this class  and throws
+       *       an assertion. Instead, use
+       *       the function reinit() to create the triangulation.
+       */
+      virtual void
+      create_triangulation(const std::vector<Point<spacedim>> &vertices,
+                           const std::vector<CellData<dim>> &  cells,
+                           const SubCellData &subcelldata) override;
+
+      /**
+       * Coarsen and refine the mesh according to refinement and coarsening
+       * flags set.
+       *
+       * @note Not implemented yet.
+       */
+      virtual void
+      execute_coarsening_and_refinement() override;
+
+      /**
+       * Override the implementation of prepare_coarsening_and_refinement from
+       * the base class.
+       *
+       * @note Not implemented yet.
+       */
+      virtual bool
+      prepare_coarsening_and_refinement() override;
+
+      /**
+       * Return true if the triangulation has hanging nodes.
+       *
+       * @note Not implemented yet.
+       */
+      virtual bool
+      has_hanging_nodes() const override;
+
+      /**
+       * Return the local memory consumption in bytes.
+       */
+      virtual std::size_t
+      memory_consumption() const override;
+
+      /**
+       * Override the function to update the number cache so we can fill data
+       * like level_ghost_owners.
+       */
+      virtual void
+      update_number_cache() override;
+
+      /**
+       * This function determines the neighboring subdomains that are adjacent
+       * to each vertex.
+       */
+      virtual std::map<unsigned int, std::set<dealii::types::subdomain_id>>
+      compute_vertices_with_ghost_neighbors() const override;
+
+      /**
+       * @copydoc dealii::parallel::DistributedTriangulationBase::is_multilevel_hierarchy_constructed
+       */
+      virtual bool
+      is_multilevel_hierarchy_constructed() const override;
+
+      /**
+       * @copydoc dealii::coarse_cell_id_to_coarse_cell_index
+       */
+      virtual unsigned int
+      coarse_cell_id_to_coarse_cell_index(
+        const types::coarse_cell_id coarse_cell_id) const override;
+
+      /**
+       * @copydoc dealii::coarse_cell_index_to_coarse_cell_id
+       */
+      virtual types::coarse_cell_id
+      coarse_cell_index_to_coarse_cell_id(
+        const unsigned int coarse_cell_index) const override;
+
+    private:
+      /**
+       * store the Settings.
+       */
+      const Settings settings;
+
+      /**
+       * Sorted list of pairs of coarse-cell ids and their indices.
+       */
+      std::vector<std::pair<types::coarse_cell_id, unsigned int>>
+        coarse_cell_id_to_coarse_cell_index_vector;
+
+      /**
+       * List of coarse-cell ids for each coarse cell (stored at cell->index()).
```

Comment:
This sounds like there may be multiple coarse-cell ids for each of the coarse cells. Can you rephrase?